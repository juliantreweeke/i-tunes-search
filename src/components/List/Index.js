import PropTypes from "prop-types";
import React from "react";
import styles from "./list.module.css";

const List = ({ data }) => {
  return (
    <ul className={styles.list}>
      {data &&
        data.map((item, index) => {
          return (
            <li key={index} className={styles.item}>
              <div className={styles.content}>
                {item.image && (
                  <img
                    src={item.image}
                    className={styles.image}
                    alt={item.title}
                  />
                )}
                <div>
                  <p>
                    <strong>{item.title}</strong>
                  </p>
                  <p>{item.text}</p>
                </div>
              </div>
              <p>{item.detail}</p>
            </li>
          );
        })}
    </ul>
  );
};

List.propTypes = {
  data: PropTypes.array,
  onCardClicked: PropTypes.func,
  refs: PropTypes.array,
};

export default List;
