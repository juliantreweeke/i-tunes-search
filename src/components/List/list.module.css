.list {
    border: 1px solid #ccc;
    border-radius: var(--radius);
}

.list .item {
    border: 1px solid var(--color-border);
    padding: var(--space-xxs) var(--space-sm);
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}
