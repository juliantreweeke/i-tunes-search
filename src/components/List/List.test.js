import React from "react";
import { render } from "@testing-library/react";
import List from "./Index";
import { axe, toHaveNoViolations } from "jest-axe";

const mockData = [
  {
    image: "testimage.com",
    url: "test.com",
    title: "title",
    text: "some text for testing",
    detail: "some detail text for testing",
  },
  {
    image: "testimage.com",
    url: "test.com",
    title: "title",
    text: "some text for testing",
    detail: "some detail text for testing",
  },
];

describe("List", () => {
  it("should render", () => {
    const { container } = render(<List data={mockData} />);
    expect(container).toBeTruthy();
  });

  it("matches snapshot with data", () => {
    const { container } = render(<List data={mockData} />);
    expect(container).toMatchSnapshot();
  });

  it("Component should have no accessibility violations", async () => {
    expect.extend(toHaveNoViolations);
    const { container } = render(<List data={mockData} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
