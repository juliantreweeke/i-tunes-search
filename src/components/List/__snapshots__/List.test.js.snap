// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List matches snapshot with data 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      <div
        class="content"
      >
        <img
          alt="title"
          class="image"
          src="testimage.com"
        />
        <div>
          <p>
            <strong>
              title
            </strong>
          </p>
          <p>
            some text for testing
          </p>
        </div>
      </div>
      <p>
        some detail text for testing
      </p>
    </li>
    <li
      class="item"
    >
      <div
        class="content"
      >
        <img
          alt="title"
          class="image"
          src="testimage.com"
        />
        <div>
          <p>
            <strong>
              title
            </strong>
          </p>
          <p>
            some text for testing
          </p>
        </div>
      </div>
      <p>
        some detail text for testing
      </p>
    </li>
  </ul>
</div>
`;
